api.c

```diff
@@ -257,16 +404,41 @@ static int api_handle_req_remove_hostfwd(Slirp *slirp, int fd,
     } else {
         struct api_hostfwd *fwd = found->data;
         const char *api_ok = "{\"return\":{}}";
-        if (slirp_remove_hostfwd(slirp, fwd->is_udp, fwd->host_addr,
-                                 fwd->host_port) < 0) {
-            const char *err = "{\"error\":{\"desc\":\"bad request: "
-                              "remove_hostfwd: slirp_remove_hostfwd failed\"}}";
-            wrc = write(fd, err, strlen(err));
-        } else {
-            ctx->hostfwds = g_list_remove(ctx->hostfwds, fwd);
-            g_free(fwd);
-            wrc = write(fd, api_ok, strlen(api_ok));
+        if (fwd->is_ipv4) {
+#if SLIRP_CONFIG_VERSION_MAX >= 3
+            if (slirp_remove_hostxfwd(slirp,
```

Comment:
same for `slirp_remove_hostxfwd`

Reply:
Checking for `SLIRP_HOSTFWD_V6ONLY` at the moment would imply `SLIRP_HOSTFWD_UDP` and the `xfwd` functions. Can we just check for that to enable the IPv6 code path?

Reply:
sure, that should work as well